backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java

```diff
@@ -480,6 +482,28 @@ public static Throwable findRootCause(Throwable throwable) {
     responseObserver.onError(statusRuntimeException);
   }
 
+  public static boolean needToRetry(Exception ex) {
```

Comment:
we need to also cover retrying on Unavailable for UAC right?

Reply:
`Unavailable for UAC` is a separate thing, we can applied same logic in AuthService & RoleService for the same.